Timelapse animation enhancements

mmcfarland opened the pull request:
## Description

* Setups of local dev environment for CORS and public azurite containers (to be paired with local Explorer development)
* Enable CORS in deployed func app
* Add optional animation api settings for progress bar and PC URL branding stamps
* Add new, optional mosaic config parameters to "hint" at per-collection animation settings. For example, you could set GOES to 15 minute frame increment, and Sentinel-2 to 2 weeks. With some collection-specific knowledge, we can have defaults that have a better chance or returning a first-time successful timelapse generation via the Explorer UI.
* Formatting and type fixes

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review
- [x] Changelog has been updated
- [x] Unit tests pass locally (./scripts/test)
- [x] Code is linted and styled (./scripts/format)

mmcfarland commented on pccommon/pccommon/tracing.py:
```diff
@@ -112,8 +112,10 @@ async def _collection_item_from_request(
         if collection_id_match:
-            collection_id = collection_id_match.group("collection_id")
-            item_id = collection_id_match.group("item_id")
+            collection_id = cast(
+                Optional[str], collection_id_match.group("collection_id")
+            )
+            item_id = cast(Optional[str], collection_id_match.group("item_id"))
```
mypy was flagging this inconsistency locally, despite not being changed in this feature.

mmcfarland commented on pcfuncs/animation/stamps/brand.py:
```diff
@@ -0,0 +1,42 @@
+
+
+text = "planetarycomputer.microsoft.com"
+font = ImageFont.truetype("/usr/share/fonts/dejavu/DejaVuSans.ttf", 12)
```
This is the only font family installed on the Azure Functions linux runtime container locally, but I need to verify it exists in the service container as well after it's deployed.

mmcfarland commented:
cc @lossyrob for review. 

Companion PR for context https://github.com/microsoft/PlanetaryComputerDataCatalog/pull/360

mmcfarland commented:
@lossrob here's an example of the URL and progress bar addition if you want to evaluate/comment.

![mspc-landsat-c2-l2-a915c758e480489ea7f8f7cb2487c518](https://user-images.githubusercontent.com/1014341/191596212-d4500ab8-3ca4-4492-99dc-cb780f0e3e82.gif)

lossyrob commented on pcfuncs/animation/stamps/brand.py:
```diff
@@ -0,0 +1,42 @@
+from PIL import Image, ImageDraw, ImageFont
```
Perhaps we should move `stamps` to `funclib` instead of `animation`. I think this would be good to include in the `image` endpoint as well, and a follow up PR could use the same logic.

mmcfarland replied:
Done in f917e5f

mmcfarland commented:
Updated to add a logo stamp, and moved stamps to funclib.

![planetarycomputer-landsat-c2-l2-dcf8d10acded49](https://user-images.githubusercontent.com/1014341/191632707-f8bef59c-aa53-4ff5-a3a0-b6d30f057fe3.gif)

mmcfarland commented on pcfuncs/funclib/stamps/stamp.py:
```diff
@@ -0,0 +1,18 @@
+
+
+class FrameStamp(ABC):
+    def __init__(self, frame: "AnimationFrame") -> None:
```
FYI, when stamps get brought into the /image endpoint, the typing here will need to change. Animations need the per-frame state.
